hamp_radar/serde_pds_files.py

```diff
@@ -0,0 +1,89 @@
+import numpy as np
+from collections.abc import Iterable
+from typing import List
+from pathlib import Path
+
+from geometries import PdsFileGeometry
+from iquick import get_geometry
+
+
+def serialize_pdsfile(filename: Path):
+    from serde.json import to_json
+
+    data = np.memmap(filename, mode="r")
+    mmbgs = list(get_geometry(data))
+    pfg = PdsFileGeometry(filename, mmbgs)
+    return to_json(pfg)
+
+
+def write_serialized_geometry(writefile: Path, geom):
+    import json
+
+    with open(writefile, "w") as json_file:
+        parsed = json.loads(geom)  # for nice formatting
+        json.dump(parsed, json_file, indent=2)
+    return parsed
+
+
+def write_pdsfile_geometries(filenames: List[Path], geomsdir: Path):
+    import time
+
+    if geomsdir.exists():
+        for filename in filenames:
+            start = time.time()
+
+            geom = serialize_pdsfile(filename)
+            writefile = geomsdir / Path(filename.name).with_suffix(".json")
+            write_serialized_geometry(writefile, geom)
+
+            end = time.time()
+            print(f"serializing {filename}: {round(end - start, 5)} s")
+    else:
+        raise ValueError("'" + str(geomsdir) + "' directory for writing doesn't exist")
+
+
+def deserialize_file_geometry(filename: Path, obj):
+    import json
+    from serde.json import from_json
+
+    with open(filename, "r") as file:
+        data = json.load(file)
+    data = json.dumps(data)
+    return from_json(obj, data)
```

Comment:
Why are you not running `from_json` directly ob the file data?

Reply:
(accident fixed)